### PR TITLE
update query version I -  provide query information to users and make query version changeable

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary

provide query information to users and make query version changeable

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://github.com/finos/legend-studio/assets/73408381/8c597605-dd07-4231-8a4c-4e90aa69fb98

https://github.com/finos/legend-studio/assets/73408381/40f11580-aad6-42a0-a918-1c0fb3606847

### CODE REVIEW

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 1]
```diff
@@ -360,6 +365,111 @@ const QueryEditorExistingQueryHeader = observer(
...
+              <div className="query-preview__field">
+                <div className="query-preview__field__label">Runtime</div>
+                <div className="query-preview__field__value">
+                  {query.runtime.value.name}
+                </div>
+              </div>
+              <div className="query-preview__field">
+                <div className="query-preview__field__label">Version</div>
+                <div className="query-preview__field__input">
+                  <InputWithInlineValidation
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/components/QueryEditor.tsx:436]: why is this not a dropdown? 
we should be able to get all the versions of the project.

**Comment by MauricioUyaguari** [comment]
when user saves initial query, the default is `latest` right, so thats what they will see in the first video?

**Comment by MauricioUyaguari** [comment]
i think we should think a bit about the user workflow. If a user updates the query version from a saved query, we should probably reload. after updating as they would be facing a different model. we should also probably add a warning if they change the version that this will happen.

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 2]
```diff
@@ -360,6 +365,111 @@ const QueryEditorExistingQueryHeader = observer(
...
+      },
+    );
+
+    const closeModal = (): void => updateState.setShowQueryInfo(false);
+
+    return (
+      <Dialog
+        open={updateState.showQueryInfo}
+        onClose={closeModal}
+        classes={{ container: 'search-modal__container' }}
```
Comment (travisstebbins) [inline packages/legend-application-query/src/components/QueryEditor.tsx:399]: Just for my own understanding, how do we decide the names/syntax used in our class names?

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 3]
```diff
@@ -360,6 +365,111 @@ const QueryEditorExistingQueryHeader = observer(
...
+                    className="btn btn--dark query-preview__field__input__action"
+                    onClick={updateQueryVersionId}
+                  >
+                    save
+                  </button>
+                </div>
+              </div>
+              {query.owner && (
+                <div className="query-preview__field">
+                  <div className="query-preview__field__label">owner</div>
```
Comment (travisstebbins) [inline packages/legend-application-query/src/components/QueryEditor.tsx:454]: Should this be capitalized like the other labels?
```suggestion
                  <div className="query-preview__field__label">Owner</div>
```

**Comment by YannanGao-gs** [comment]
> when user saves initial query, the default is `latest` right, so thats what they will see in the first video?

I think it depends on how you create a query.
Currently, i just chose to `create a new query on a mapping`

Screenshot 2024-02-13 at 1 53 08 PM

**Comment by YannanGao-gs** [comment]
> when user saves initial query, the default is `latest` right, so thats what they will see in the first video?

The first video is about displaying query's information e.g. project id, its current version id, owner..

File: packages/legend-application-query/src/components/QueryEditor.tsx (hunk 2 above)
Comment (YannanGao-gs) [reply packages/legend-application-query/src/components/QueryEditor.tsx:426]: I just reuse some shared modals in the codebase

**Comment by YannanGao-gs** [comment]
> i think we should think a bit about the user workflow. If a user updates the query version from a saved query, we should probably reload. after updating as they would be facing a different model. we should also probably add a warning if they change the version that this will happen.


query now will be reloaded once users click the 'save' button to update version


To enable revert to a working version, I think I need to update the engine/depot to record the original version ID that this query is created at, which will be implemented in another PR

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**